pkg/proxy/proxyresponse/types.go

```diff
@@ -0,0 +1,58 @@
+package proxyresponse
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+)
+
+// BackendResponse wraps an HTTP response with metadata about the backend
+type BackendResponse struct {
+	Response    *http.Response
+	BackendName string
+	BackendURL  string
+}
+
+// BackendError wraps an error with metadata about which backend caused it
+type BackendError struct {
+	Err         error
+	BackendName string
+	BackendURL  string
+}
+
+// ForwardBackendError sends a simple error response: {backend}: {error}
+func ForwardBackendError(w http.ResponseWriter, backendName string, statusCode int, bodyBytes []byte, logger log.Logger) {
+	level.Error(logger).Log(
+		"msg", "Forwarding backend error to client",
+		"backend", backendName,
+		"status", statusCode,
+	)
+
+	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
+	w.WriteHeader(statusCode)
```

Comment:
I wonder if we could not just simplify this, and instead of `w.Header().Set("Content-Type", "text/plain; charset=utf-8")`, add the backend here something like:
```go
	w.Header().Set("Failed-Backend", backendName)
```